Index: refpolicy-ubuntu-0.2.20091117/policy/modules/services/policykit.fc
===================================================================
--- refpolicy-ubuntu-0.2.20091117.orig/policy/modules/services/policykit.fc	2010-04-22 17:31:46.360393273 -0700
+++ refpolicy-ubuntu-0.2.20091117/policy/modules/services/policykit.fc	2010-04-22 17:45:08.640564552 -0700
@@ -8,8 +8,12 @@
 /usr/libexec/polkit-resolve-exe-helper.* --	gen_context(system_u:object_r:policykit_resolve_exec_t,s0)
 /usr/libexec/polkitd			--	gen_context(system_u:object_r:policykit_exec_t,s0)
 
+/usr/lib/policykit-1/polkit-agent-helper-1 	--	gen_context(system_u:object_r:policykit_resolve_exec_t,s0)
+/usr/lib/policykit-1/polkitd			--	gen_context(system_u:object_r:policykit_exec_t,s0)
+
 /var/lib/misc/PolicyKit.reload			gen_context(system_u:object_r:policykit_reload_t,s0)
 /var/lib/PolicyKit(/.*)?			gen_context(system_u:object_r:policykit_var_lib_t,s0)
 /var/lib/PolicyKit-public(/.*)?			gen_context(system_u:object_r:policykit_var_lib_t,s0)
 /var/run/PolicyKit(/.*)?			gen_context(system_u:object_r:policykit_var_run_t,s0)
 
+/var/lib/polkit-1(/.*)?			gen_context(system_u:object_r:policykit_var_lib_t,s0)
Index: refpolicy-ubuntu-0.2.20091117/policy/modules/services/policykit.if
===================================================================
--- refpolicy-ubuntu-0.2.20091117.orig/policy/modules/services/policykit.if	2010-04-22 17:31:46.380394360 -0700
+++ refpolicy-ubuntu-0.2.20091117/policy/modules/services/policykit.if	2010-04-22 17:45:08.640564552 -0700
@@ -23,6 +23,24 @@
 
 ########################################
 ## <summary>
+##	Execute a domain transition to run polkit
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`policykit_domtrans',`
+	gen_require(`
+		type policykit_t, policykit_exec_t;
+	')
+
+	domtrans_pattern($1, policykit_exec_t, policykit_t)
+')
+
+########################################
+## <summary>
 ##	Execute a domain transition to run polkit_auth.
 ## </summary>
 ## <param name="domain">
@@ -172,6 +190,24 @@
 
 ########################################
 ## <summary>
+##	Read polkit pid files.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`policykit_read_pid_files',`
+	gen_require(`
+		type policykit_var_run_t;
+	')
+
+	read_files_pattern($1, policykit_var_run_t, policykit_var_run_t)
+')
+
+########################################
+## <summary>
 ##	Search policykit lib directories.
 ## </summary>
 ## <param name="domain">
Index: refpolicy-ubuntu-0.2.20091117/policy/modules/services/policykit.te
===================================================================
--- refpolicy-ubuntu-0.2.20091117.orig/policy/modules/services/policykit.te	2010-04-22 17:31:46.360393273 -0700
+++ refpolicy-ubuntu-0.2.20091117/policy/modules/services/policykit.te	2010-04-22 17:45:08.680423838 -0700
@@ -36,8 +36,8 @@
 # policykit local policy
 #
 
-allow policykit_t self:capability { setgid setuid };
-allow policykit_t self:process getattr;
+allow policykit_t self:capability { setgid setuid sys_ptrace };
+allow policykit_t self:process { getattr getsched };
 allow policykit_t self:fifo_file rw_file_perms;
 allow policykit_t self:unix_dgram_socket create_socket_perms;
 allow policykit_t self:unix_stream_socket create_stream_socket_perms;
@@ -70,6 +70,30 @@
 
 userdom_read_all_users_state(policykit_t)
 
+fs_list_inotifyfs(policykit_t)
+
+optional_policy(`
+	dbus_system_bus_client(policykit_t)
+	dbus_session_bus_client(policykit_t)
+	dbus_connect_system_bus(policykit_t)
+
+	optional_policy(`
+		consolekit_dbus_chat(policykit_t)
+	')
+
+	optional_policy(`
+		networkmanager_dbus_chat(policykit_t)
+	')
+
+	optional_policy(`
+		consolekit_dbus_chat(policykit_t)
+	')
+
+	optional_policy(`
+		unconfined_dbus_chat(policykit_t)
+	')
+')
+
 ########################################
 #
 # polkit_auth local policy
