---
 policy/modules/system/udev.te |   21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

Index: refpolicy-ubuntu-0.2.20091117/policy/modules/system/udev.te
===================================================================
--- refpolicy-ubuntu-0.2.20091117.orig/policy/modules/system/udev.te	2010-04-22 17:31:46.800570026 -0700
+++ refpolicy-ubuntu-0.2.20091117/policy/modules/system/udev.te	2010-04-22 17:45:08.460466602 -0700
@@ -158,6 +158,7 @@
 sysnet_signal_dhcpc(udev_t)
 sysnet_manage_config(udev_t)
 sysnet_etc_filetrans_config(udev_t)
+sysnet_manage_network_state(udev_t)
 
 userdom_dontaudit_search_user_home_content(udev_t)
 
@@ -187,6 +188,26 @@
 	')
 ')
 
+ifdef(`distro_ubuntu',`
+	fs_manage_tmpfs_dirs(udev_t)
+	fs_manage_tmpfs_files(udev_t)
+	fs_manage_tmpfs_symlinks(udev_t)
+	fs_manage_tmpfs_sockets(udev_t)
+	fs_manage_tmpfs_blk_files(udev_t)
+	fs_manage_tmpfs_chr_files(udev_t)
+	fs_relabel_tmpfs_blk_file(udev_t)
+	fs_relabel_tmpfs_chr_file(udev_t)
+
+	term_search_ptys(udev_t)
+
+	# for arping used for static IP addresses on PCMCIA ethernet
+	netutils_domtrans(udev_t)
+
+	optional_policy(`
+		unconfined_domain(udev_t)
+	')
+')
+
 optional_policy(`
 	alsa_domtrans(udev_t)
 	alsa_read_lib(udev_t)
