---
 policy/users |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: refpolicy/policy/users
===================================================================
--- refpolicy.orig/policy/users
+++ refpolicy/policy/users
@@ -30,7 +30,7 @@ gen_user(staff_u, staff, staff_r sysadm_
 gen_user(sysadm_u, sysadm, sysadm_r, s0, s0 - mls_systemhigh, mcs_allcats)
 
 # Until order dependence is fixed for users:
-gen_user(unconfined_u, unconfined, unconfined_r, s0, s0 - mls_systemhigh, mcs_allcats)
+gen_user(unconfined_u, unconfined, unconfined_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
 
 #
 # The following users correspond to Unix identities.
